### ISSUE TITLE
Model hash is always the same

### ISSUE THREAD

**Issue by dali99** [d8ahazard/sd_dreambooth_extension#134]
It seems a little weird to me that the model hash is always the same, I would have expected it to change with the training steps.

But I dont know what exactly goes into the calculation of the model hash.

**Comment by d8ahazard** [comment]
It definitely should be different for every step.

What base checkpoint are you using? I'd suggest trying with the 1.5 checkpoint, if even for a few steps. It changes on mine.

**Comment by dali99** [comment]
I am using v1-5-pruned-emaonly as a base.

Maybe it's some weird thing with pytorch-rocm, but it'd be weird since that'd be the only thing broken (and I wouldn't expect it to be a significant part of getting a hash)

**Comment by dali99** [comment]
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/3324
might be related?

**Comment by d8ahazard** [comment]
Possibly. All I can say is from what I've tested, and my hash always changes. What if you use the non-pruned version?

**Comment by dali99** [comment]
Can confirm nothing changes when using v1-5-pruned (which I think is what you meant?)

**Comment by dali99** [comment]
updating my stuff (possibly changing pytorch version) changed the hash for newer models, but these new models still have the same hash

**Comment by d8ahazard** [comment]
Could you provide screenshot of all your training parameters. 

Something isn't right here, and I suspect it's not the code. I regularly train on the "normal" checkpoint with no issues. Only time I had the same hash over and over was when I was failing to convert the VAE or UNET, which I've since fixed.

**Comment by cheezecrisp** [comment]
I have trained a model for 3000 steps, the pictures it generates are IDENTICAL to source checkpoint (Using same seed/conditions).
what's wrong?

**Comment by dali99** [comment]
@cheezecrisp make a new issue, this issue is for tracking when training is working, but the model hash is not updated (other than changing like once since it's not identical to the source checkpoint).

@d8ahazard 
![Screen Shot 2022-11-14 at 17 59 58](https://user-images.githubusercontent.com/990767/201720657-5f65e852-7e51-4f37-b589-28e4b6e1dc0a.png)

here's one I'm currently doing, but I've done a lot of different configurations, I have seven very different models spread across 2 model hashes

**Comment by d8ahazard** [comment]
Do you get any errors in the console when it tries to save a checkpoint? Or, does the UI report that it saved the checkpoint somewhere?

More importantly, have you ever used an app like beyond compare before? 

Basically, what I want to know is if any of the files in /models/dreambooth/agnjdhz/working are updated after you do any number of training steps.

I'd make a copy of working, train for 5 steps, and then diff working and working(copy) to see if any of the .bin files have changed. At the very least, I believe the vae and unet should change, or maybe just the unet. But at least one of the .bin files int here should be updated.

If not, I'd expect some kind of errors in the log when it tries to save a checkpoint.

**Comment by Art4DD** [comment]
I can confirm this error.
I have the same thing:
![image](https://user-images.githubusercontent.com/55113758/202054702-2e79c482-a1a4-4e77-be9e-bd25852c660c.png)
I don't get any errors in console.
.bin files in /models/dreambooth/ are overwritten when saving the model, but have the same weight in bytes.

**Comment by dali99** [comment]
It seems like when training the text encoder it does change hash, but with it off it does not change the hash.

**Comment by Slug-Cat** [comment]
My hashes are also identical. It makes things difficult when trying to retrieve old gen parameters, as they are listed as only a hash.

**Comment by d8ahazard** [comment]
Try with the latest version, please. I've set it to save the VAE as well if it's used.

**Comment by dali99** [comment]
Can confirm this is still an issue

**Comment by knoopx** [comment]
FYI webui hashing algorithm is worthless, it only reads a few kbs to hash the model, dont rely on it for anything serious. There's nothing wrong with this extension. https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/828438b4a190759807f9054932cae3a8b880ddf1/modules/sd_models.py#L94

**Comment by mykeehu** [comment]
But the [hash problem has persisted](https://github.com/d8ahazard/sd_dreambooth_extension/issues/386) since then, which is a problem because that's how SD identifies the model. Multiple backups with the same hash do not bode well. I'm waiting for the fix, until then I won't train anymore, because I can't generate new hashes afterwards.
